### ISSUE TITLE
Tunnelling when resolving circle->line collision

### ISSUE THREAD

**Issue by Alloyed** [bakpakin/Splash.lua#5]
Hi, I've been using splash (even in its unfinished state) for a while now, but I think this is the first "real" bug I've found. Specifically, the circle->line sweep seems to occasionally miss collisions.

I can pretty reliably reproduce it inside my game by firing an arrow (a circle moving at an arc) at a vertical wall (line, 90 degrees) from the right. I haven't experienced it yet from other kinds of collisions, including against the same wall from the left. Here's a webm to demonstrate:
https://streamable.com/ca1ox

I'll try to make a proper test harness, or maybe even fix the bug myself in the future, but I figured I'd post early to see if you're still interested in this repo.

**Comment by bakpakin** [comment]
Yes, some of these collisions were not perfect. If you could fix them, that would be awesome. Most of the bugs seem to be with line segments and circles.

I suspect floating point inaccuracy is at play, or maybe 0 and infinity handling.
